components/rpill/common/rpill.cc

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/rpill/common/rpill.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/barrier_closure.h"
+#include "base/callback.h"
+#include "base/callback_helpers.h"
+#include "base/strings/string_util.h"
+#include "base/system/sys_info.h"
+
+namespace brave_rpill {
+
+namespace {
+
+struct DeviceNameInfo {
+  std::string manufacturer_name;
+  std::string model_name;
+};
+
+std::string BuildDeviceId(DeviceNameInfo* info) {
+  return base::ToLowerASCII(info->manufacturer_name + info->model_name);
+}
+
+bool IsUncertainFuture(DeviceNameInfo* info) {
+  const std::string device_id = BuildDeviceId(info);
+
+  const std::vector<std::string> keywords = {"amazon", "virtualbox", "vmware",
+                                             "xen"};
+
+  for (const auto& keyword : keywords) {
+    if (device_id.find(keyword) != std::string::npos) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+void OnDeviceNameInfoReady(IsUncertainFutureCallback callback,
+                           std::unique_ptr<DeviceNameInfo> info) {
+  const bool is_uncertain_future = IsUncertainFuture(info.get());
+  std::move(callback).Run(is_uncertain_future);
+}
+
+void OnHardwareInfoReady(DeviceNameInfo* device_name_info_ptr,
+                         base::ScopedClosureRunner done_closure,
+                         base::SysInfo::HardwareInfo hardware_info) {
+  device_name_info_ptr->manufacturer_name =
```

Comment:
If you decide not to make the functions i mentioned before methods on `DeviceNameInfo`, you can instead make device_name_info_ptr a `std::unique_ptr<base::SysInfo::HardwareInfo>` since these structs are the same.

https://source.chromium.org/chromium/chromium/src/+/master:base/system/sys_info.h;l=85-88?q=HardwareInfo&ss=chromium

Reply:
Using our own struct here so we can expand at a later time as we will need the architecture for some future checks. Thanks